version: '3.8'

services:
    worker1:
      build: .
      command: uvicorn src.referalbot.api.main:app --host 0.0.0.0 --port 8000
      ports:
        - "8000:8000" 
      environment:
      - DB_HOST=db
      - DB_PORT=5432
      - DB_NAME=${DB_PATH}
      - DB_USER=${DB_USER}
      - DB_PASSWORD=${DB_PASSWORD}
      - DB_SYSTEM=postgresql
      - DB_DRIVER=asyncpg
      - DATABASE_URL=${DATABASE_URL}
      env_file: .env.docker
      restart: unless-stopped
      depends_on:
        db:
          condition: service_healthy
      logging:
        driver: "json-file"
        options:
          max-size: "10m"
          max-file: "3"
          tag: "parser-api"

    worker2:
      build: .
      command: python -m src.referalbot.bot.main
      environment:
      - DB_HOST=db
      - DB_PORT=5432
      - DB_NAME=${DB_PATH}
      - DB_USER=${DB_USER}
      - DB_PASSWORD=${DB_PASSWORD}
      - DB_SYSTEM=postgresql
      - DB_DRIVER=asyncpg
      - DATABASE_URL=${DATABASE_URL}
      env_file: .env.docker
      restart: unless-stopped
      depends_on:
        db:
          condition: service_healthy
      logging:
        driver: "json-file"
        options:
          max-size: "10m"
          max-file: "3"
          tag: "parser-api"


    db:
      image: postgres:15-alpine
      environment:
        POSTGRES_DB: ${DB_PATH}
        POSTGRES_USER: ${DB_USER}
        POSTGRES_PASSWORD: ${DB_PASSWORD}
      env_file:
        - .env.docker
      restart: unless-stopped
      healthcheck:
        test: ["CMD-SHELL", "pg_isready -U ${DB_USER} -d ${DB_PATH}"]
        interval: 5s
        timeout: 5s
        retries: 5
      logging:
        driver: "json-file"
        options:
          max-size: "10m"
          max-file: "3"